### PR TITLE
changelog: add 1.17.0 CHANGELOG

### PR DESCRIPTION
**Pull request by imurchie**
Beginnings of 1.17.0 CHANGELOG. Please add updates in comments, if you want (with PRs for the change)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,43 @@
...
+  * Add `waitForWebviewMs` option to `mobile:getContexts` endpoint, to block for a period while waiting for webview report (https://github.com/appium/appium-xcuitest-driver/pull/1135)
+  * Allow SafariViewController webviews to be reached automatically (https://github.com/appium/appium-remote-debugger/pull/189) (https://github.com/appium/appium-remote-debugger/pull/197)
+  * Fix asynchronous execution in webview context for iOS 12.1 and below (https://github.com/appium/appium-remote-debugger/pull/199)
+  * Fix handling of React components whose state is handled by React (https://github.com/appium/appium-remote-debugger/pull/202)
+  * Fix webviews for iOS 13.3 and 13.4 (https://github.com/appium/appium-remote-debugger/pull/203) (https://github.com/appium/appium-remote-debugger/pull/204) (https://github.com/appium/appium-remote-debugger/pull/205) (https://github.com/appium/appium-remote-debugger/commit/e067f0eab012b5ef67f1ce67eb85479c4abe440b) (https://github.com/appium/appium-xcuitest-driver/pull/1174)
+  * Fix handling of alerts opened from webview (https://github.com/appium/appium-xcuitest-driver/pull/1176)
+
+#### Android General
```
Comment (KazuCocoa) [inline CHANGELOG.md]: - Add `mobile:` functions:
    - `mobile: listSms` to get the list of SMS (https://github.com/appium/appium-android-driver/pull/602 https://github.com/appium/appium-espresso-driver/pull/548)
    - `mobile: getNotifications` to get the list of notifications (https://github.com/appium/appium-android-driver/pull/598)
- Fix `toggle_location_services` on Android 9.0 (https://github.com/appium/appium-android-driver/pull/594)
- Add to allow letter as a `platformVersion` like `R` in preview/beta Android OS release ( https://github.com/appium/appium-android-driver/pull/607 )

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,43 @@
...
+
+#### Android (UiAutomator2)
+
+#### Android (Espresso)
```
Comment (KazuCocoa) [inline CHANGELOG.md]: - Add `espressoBuildConfig` capability to customize the espresso server build configuration (https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md#espresso-server-build-configuration-json)
- Add `-android viewmatcher` selector (https://github.com/appium/appium-espresso-driver/pull/516)
- Add `mobile:` functions:
    - `mobile: startScreenStreaming`, `mobile: stopScreenStreaming` (https://github.com/appium/appium-espresso-driver/pull/521)
- Allow to conduct `mobile:` functions, perform actions,  on WebView context (https://github.com/appium/appium-espresso-driver/pull/549)
- Add `noMultilineButtons`, `noEllipsizedText` and `noOverlaps` as available attributes as element attributes to help view assertions (https://github.com/appium/appium-espresso-driver/pull/544)
- Add `intentOptions` to customize intent for launching activities (https://github.com/appium/appium-espresso-driver/pull/542)
- Add `skipServerInstallation` capability to skip espresso driver preparation (https://github.com/appium/appium-espresso-driver/pull/526)
- Support `open_notifications` endpoint (https://github.com/appium/appium-espresso-driver/pull/536/files)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,43 @@
...
+  * Add `waitForWebviewMs` option to `mobile:getContexts` endpoint, to block for a period while waiting for webview report (https://github.com/appium/appium-xcuitest-driver/pull/1135)
+  * Allow SafariViewController webviews to be reached automatically (https://github.com/appium/appium-remote-debugger/pull/189) (https://github.com/appium/appium-remote-debugger/pull/197)
+  * Fix asynchronous execution in webview context for iOS 12.1 and below (https://github.com/appium/appium-remote-debugger/pull/199)
+  * Fix handling of React components whose state is handled by React (https://github.com/appium/appium-remote-debugger/pull/202)
+  * Fix webviews for iOS 13.3 and 13.4 (https://github.com/appium/appium-remote-debugger/pull/203) (https://github.com/appium/appium-remote-debugger/pull/204) (https://github.com/appium/appium-remote-debugger/pull/205) (https://github.com/appium/appium-remote-debugger/commit/e067f0eab012b5ef67f1ce67eb85479c4abe440b) (https://github.com/appium/appium-xcuitest-driver/pull/1174)
+  * Fix handling of alerts opened from webview (https://github.com/appium/appium-xcuitest-driver/pull/1176)
```
Comment (KazuCocoa) [inline CHANGELOG.md]: - Add `screenshotOrientation` as Settings API to customize screenshot orientation (https://github.com/appium/WebDriverAgent/pull/277)

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,43 @@
...
+
+#### Android General
+
+#### Android (UiAutomator2)
+
+#### Android (Espresso)
+
+#### Windows
+
```
Comment (KazuCocoa) [inline CHANGELOG.md]: ## macOS
- Added `a4mHost`, `a4mPort`, `a4mAppPath` and `killAllA4MAppBeforeStart` capabilities to make mac driver configurable
(https://github.com/appium/appium-mac-driver/pull/42, https://github.com/appium/appium-mac-driver/pull/44)

File: CHANGELOG.md (hunk 1 above)
Comment (mykola-mokhnach) [reply CHANGELOG.md]: Updated emulator detection logic: https://github.com/appium/appium/pull/14015#discussion_r386749006

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,43 @@
...
+  * Allow SafariViewController webviews to be reached automatically (https://github.com/appium/appium-remote-debugger/pull/189) (https://github.com/appium/appium-remote-debugger/pull/197)
+  * Fix asynchronous execution in webview context for iOS 12.1 and below (https://github.com/appium/appium-remote-debugger/pull/199)
+  * Fix handling of React components whose state is handled by React (https://github.com/appium/appium-remote-debugger/pull/202)
+  * Fix webviews for iOS 13.3 and 13.4 (https://github.com/appium/appium-remote-debugger/pull/203) (https://github.com/appium/appium-remote-debugger/pull/204) (https://github.com/appium/appium-remote-debugger/pull/205) (https://github.com/appium/appium-remote-debugger/commit/e067f0eab012b5ef67f1ce67eb85479c4abe440b) (https://github.com/appium/appium-xcuitest-driver/pull/1174)
+  * Fix handling of alerts opened from webview (https://github.com/appium/appium-xcuitest-driver/pull/1176)
+
+#### Android General
+
+#### Android (UiAutomator2)
+
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: - Added `mobile: type` endpoint: https://github.com/appium/appium-uiautomator2-driver/pull/365
- Made `mobile:` commands available in web context: https://github.com/appium/appium-uiautomator2-driver/pull/366
- Tuned `sendKeys` endpoint: https://github.com/appium/appium-uiautomator2-server/pull/329

File: CHANGELOG.md (hunk 2 above)
Comment (mykola-mokhnach) [reply CHANGELOG.md]: `mobile: startScreenStreaming, mobile: stopScreenStreaming` have been added to all android drivers

File: CHANGELOG.md (hunk 4 above)
Comment (mykola-mokhnach) [reply CHANGELOG.md]: Added a possibility to autodetect the location of the server app: https://github.com/appium/appium-mac-driver/pull/44

File: CHANGELOG.md (hunk 2 above)
Comment (mykola-mokhnach) [reply CHANGELOG.md]: > Allow to conduct mobile: functions, perform actions, on WebView context (appium/appium-espresso-driver#549)

This has also been done for UIA2

**Comment by imurchie** [comment]
Thanks @KazuCocoa  and @mykola-mokhnach !

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,69 @@
+CHANGES IN VERSION 1.17.0 (FROM 1.16.0)
+===================================
+
+Appium 1.17.0 is a minor release.
+
+#### General
+* Add `mobile:` functions:
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: this is only for Android

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,69 @@
...
+  * `mobile: startScreenStreaming`, `mobile: stopScreenStreaming` (https://github.com/appium/appium-espresso-driver/pull/521)
+
+#### iOS General
+
+#### iOS (XCUITest)
+* Fix app installation on real devices (https://github.com/appium/appium-xcuitest-driver/pull/1139) (https://github.com/appium/appium-xcuitest-driver/pull/1138) (https://github.com/appium/appium-xcuitest-driver/pull/1140) (https://github.com/appium/appium-xcuitest-driver/pull/1152) (https://github.com/appium/appium-ios-device/pull/67)
+* Fix temporary log file to be more efficient (https://github.com/appium/appium-xcuitest-driver/pull/1147)
+* Fix closing of proxies during shutdown (https://github.com/appium/appium-xcuitest-driver/pull/1153)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: I'd rather say `fixed ports cleanup`

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,69 @@
...
+* Capabilities:
+  * `simulatorStartupTimeout` change the default timeout for simulator startup (https://github.com/appium/appium-xcuitest-driver/pull/1163)
+  * `simulatorPasteboardAutomaticSync` turn on/off simulator pasteboard synching at launch (https://github.com/appium/appium-xcuitest-driver/pull/1168)
+  * `simulatorTracePointer` turn on/off pointer highlighting in simulators (https://github.com/appium/appium-xcuitest-driver/pull/1169)
+* Add `mobile:` functions for new `simctl` functionality (https://github.com/appium/appium-xcuitest-driver/pull/1162) (https://github.com/appium/appium-ios-simulator/pull/260) (https://github.com/appium/node-simctl/pull/96)
+  * `mobile: setPermission`
+  * `mobile: getAppearance`
+  * `mobile: setAppearance`
+* Webviews/Safari:
+  * Add webview bundle identifier to data returned with `fullContextList`
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: it probably also make sense to mention these new caps for extended remote debugger logging

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,69 @@
...
+* Added `mobile: type` endpoint (https://github.com/appium/appium-uiautomator2-driver/pull/365)
+* Tuned `sendKeys` endpoint (https://github.com/appium/appium-uiautomator2-server/pull/329)
+
+#### Android (Espresso)
+* Capabilities:
+  * `espressoBuildConfig` customize the espresso server build configuration (https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md#espresso-server-build-configuration-json)
+  * `intentOptions` customize intent for launching activities (https://github.com/appium/appium-espresso-driver/pull/542)
+  * `skipServerInstallation` skip espresso driver preparation (https://github.com/appium/appium-espresso-driver/pull/526)
+* Add `-android viewmatcher` selector (https://github.com/appium/appium-espresso-driver/pull/516)
+* Add `noMultilineButtons`, `noEllipsizedText` and `noOverlaps` as available attributes as element attributes to help view assertions (https://github.com/appium/appium-espresso-driver/pull/544)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: `as available element attributes`

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium/pull/14021 is a PR to 1.17 branch in order to update dependencies to apply all of fixes addressed in this PR.
(e.g. Fixes for macOS notarization, up to date remote-debugger)

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,69 @@
+CHANGES IN VERSION 1.17.0 (FROM 1.16.0)
+===================================
+
+Appium 1.17.0 is a minor release.
+
+#### General
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: One thing to put into General:
- Active driver sessions are now properly cleanup upon main Appium process termination: https://github.com/appium/appium/pull/13913

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,69 @@
...
+* Fix temporary log file to be more efficient (https://github.com/appium/appium-xcuitest-driver/pull/1147)
+* Fix closing of proxies during shutdown (https://github.com/appium/appium-xcuitest-driver/pull/1153)
+* Fix starting of `idb` when used (https://github.com/appium/appium-idb/pull/20)
+* Add `screenshotOrientation` as Settings API to customize screenshot orientation (https://github.com/appium/WebDriverAgent/pull/277)
+* Capabilities:
+  * `simulatorStartupTimeout` change the default timeout for simulator startup (https://github.com/appium/appium-xcuitest-driver/pull/1163)
+  * `simulatorPasteboardAutomaticSync` turn on/off simulator pasteboard synching at launch (https://github.com/appium/appium-xcuitest-driver/pull/1168)
+  * `simulatorTracePointer` turn on/off pointer highlighting in simulators (https://github.com/appium/appium-xcuitest-driver/pull/1169)
+* Add `mobile:` functions for new `simctl` functionality (https://github.com/appium/appium-xcuitest-driver/pull/1162) (https://github.com/appium/appium-ios-simulator/pull/260) (https://github.com/appium/node-simctl/pull/96)
```
Comment (dpgraham) [inline CHANGELOG.md]: I would add to this `for new `simctl` functionality in Xcode 11.4+`

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,69 @@
...
+  * `mobile: startScreenStreaming`, `mobile: stopScreenStreaming` (https://github.com/appium/appium-espresso-driver/pull/521)
+
+#### iOS General
+
+#### iOS (XCUITest)
+* Fix app installation on real devices (https://github.com/appium/appium-xcuitest-driver/pull/1139) (https://github.com/appium/appium-xcuitest-driver/pull/1138) (https://github.com/appium/appium-xcuitest-driver/pull/1140) (https://github.com/appium/appium-xcuitest-driver/pull/1152) (https://github.com/appium/appium-ios-device/pull/67)
+* Fix temporary log file to be more efficient (https://github.com/appium/appium-xcuitest-driver/pull/1147)
+* Fix closing of proxies during shutdown (https://github.com/appium/appium-xcuitest-driver/pull/1153)
+* Fix starting of `idb` when used (https://github.com/appium/appium-idb/pull/20)
+* Add `screenshotOrientation` as Settings API to customize screenshot orientation (https://github.com/appium/WebDriverAgent/pull/277)
```
Comment (KazuCocoa) [inline CHANGELOG.md]: I found `snapshotMaxDepth` https://github.com/appium/appium/pull/14028

- Add `snapshotMaxDepth` as Settings API to restrict the depth of getting the elements source tree (https://github.com/appium/WebDriverAgent/pull/273)

File: CHANGELOG.md (hunk 8 above)
Comment (imurchie) [reply CHANGELOG.md]: They are in the entry for 1.16.0

File: CHANGELOG.md [hunk 13]
```diff
@@ -51,7 +53,7 @@ Appium 1.17.0 is a minor release.
   * `intentOptions` customize intent for launching activities (https://github.com/appium/appium-espresso-driver/pull/542)
   * `skipServerInstallation` skip espresso driver preparation (https://github.com/appium/appium-espresso-driver/pull/526)
 * Add `-android viewmatcher` selector (https://github.com/appium/appium-espresso-driver/pull/516)
-* Add `noMultilineButtons`, `noEllipsizedText` and `noOverlaps` as available attributes as element attributes to help view assertions (https://github.com/appium/appium-espresso-driver/pull/544)
+* Add `noMultilineButtons`, `noEllipsizedText` and `noOverlaps` as available attributes as element attributes to extend available view assertions (https://github.com/appium/appium-espresso-driver/pull/544)
```
Comment (mykola-mokhnach) [inline CHANGELOG.md]: `as available attributes as element attributes` does not sound nice %)

**Pull request merged**